docs/development/module/fees.md

```diff
@@ -0,0 +1,209 @@
+magnitude, and thus Substrate provides a flexible mechanism for characterizing the total, presumably
+minimum, cost of a transaction in order to be included into a block.
+
+The fee system is heavily linked to the [weight system](conceptual/runtime/weight.md). Make sure to read and understand weights
+section before continuing this document.
+
+## Transaction Fees
+
+The fee to include a transaction consists of three parts:
+
+* `base_fee` a fixed fee that is applied to every single transaction. See
+  [`TransactionBaseFee`](/rustdocs/master/srml_transaction_payment/trait.Trait.html#associatedtype.TransactionBaseFee)
+* `length_fee` a per-byte fee that is multiplied by the length, in bytes, of the encoded
+  transaction. See
+  [`TransactionByteFee`](/rustdocs/master/srml_transaction_payment/trait.Trait.html#associatedtype.TransactionByteFee)
+* `weight_fee` a per-weight-unit fee that is multiplied by the weight of the transaction. As
+  mentioned, weight of each dispatch is denoted via the flexible `#[weight]` annotation. Knowing the
+  weight, it must be converted to a deductible `balance` type (typically denoted by a module that
+  implements `Currency`, `srml-balances` in substrate node). For this, each runtime must define a
+  [`WeightToFee`](/rustdocs/master/srml_transaction_payment/trait.Trait.html#associatedtype.WeightToFee)
+  type that makes the conversion. `WeightToFee` must be a struct that implements a [`Convert<Weight,
+  Balance>`](/rustdocs/master/sr_primitives/traits/trait.Convert.html).
+
+based on the above, the final fee of a dispatchable is:
+
+```
+fee =
+  base_fee +
+  len(tx) * length_fee +
+  WeightToFee(weight)
+```
+
+Customizing the above would be as simple as configuring the appropriate associated type in the
+respective module.
+
+```rust
+use sr_primitives::{traits::Convert, weights::Weight}
+// Assume this is the balance type
+type Balance = u64;
+
+// Assume we want all the weights to have a `100 + 2 * w` conversion to fees
+struct CustomWeightToFee;
+impl Convert<Weight, Balance> for CustomWeightToFee {
+    fn convert(w: Weight) -> Balance {
+        let a = Balance::from(100);
+        let b = Balance::from(2);
+        let w = Balance::from(w);
+        a + b * w
+    }
+}
+
+parameter_types! {
+    TransactionBaseFee: Balance = 10;
+    TransactionByteFee: Balance = 10;
+}
+
+impl transaction_payment::Trait {
+    TransactionBaseFee = TransactionBaseFee;
+    TransactionByteFee = TransactionByteFee;
+    WeightToFee = CustomWeightToFee;
+    // we will get to this one soon enough
+    FeeMultiplierUpdate = ();
+}
+
+```
+
+Two further questions need to be answered, given the above snippet.
+
+1. What is the purpose of `FeeMultiplierUpdate`?
+2. Knowing how we are capable of defining the conversion of weights to fees (via `WeightToFee`), the
+   question remains, how can we define and customize the weights in the first place?
+
+We will answer each question in the following sections, respectively.
```

Comment:
I really don't like this style of writing, sorry. You should get a second opinion though.

Reply:
Appreciated. I read it again and I liked it tbh. Don't know what to do but open to edit.